### PR TITLE
Fixed #28749 -- Added subquery support for PostgreSQL’s ArrayField 'in' lookup

### PR DESCRIPTION
**Pull request by mpasternak**
https://code.djangoproject.com/ticket/28749#ticket

### CODE REVIEW

**Comment by mpasternak** [comment]
Hi @adamchainz , updated as suggested. As I understand, CI checks are being triggered automatically, right?

**Comment by adamchainz** [comment]
Yes there might just be a queue though

File: docs/releases/2.0.txt [hunk 1]
```diff
@@ -163,6 +163,9 @@ Minor features
 * :djadmin:`inspectdb` can now introspect ``JSONField`` and various
   ``RangeField``\s (``django.contrib.postgres`` must be in ``INSTALLED_APPS``).
 
+* The ``in`` lookup for `:class:`~django.contrib.postgres.fields.array.ArrayField`
+  supports subqueries
```
Comment (adamchainz) [inline docs/releases/2.0.txt]: The ` before :class: is invalid
Add a period/full stop
📝🚀

File: tests/postgres_tests/test_array.py [hunk 2]
```diff
@@ -176,6 +175,17 @@ def test_in(self):
             self.objs[:2]
         )
 
+    def test_in_subquery(self):
+        IntegerArrayModel.objects.create(field=[2, 3])
+
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.filter(
+                field__in=IntegerArrayModel.objects.all().values_list(
+                    "field", flat=True)
```
Comment (adamchainz) [inline tests/postgres_tests/test_array.py]: this is probably cleaner as one line

**Comment by mpasternak** [comment]
Commited, thanks

File: docs/releases/2.0.txt [hunk 3]
```diff
@@ -163,6 +163,9 @@ Minor features
 * :djadmin:`inspectdb` can now introspect ``JSONField`` and various
   ``RangeField``\s (``django.contrib.postgres`` must be in ``INSTALLED_APPS``).
 
+* The ``in`` lookup for :class:`~django.contrib.postgres.fields.array.ArrayField`
+  supports subqueries
```
Comment (adamchainz) [inline docs/releases/2.0.txt]: Last thing I can see, you still need to add a period (.) at the end of the sentence 😉

**Comment by adamchainz** [comment]
Still a couple CI failures

doc:

```
/home/jenkins/workspace/docs/docs/releases/2.0.txt:166: WARNING: py:class reference target not found: django.contrib.postgres.fields.array.ArrayField
```

Looks like this needs fixing, should be simple

isort:

```
+++ /home/jenkins/workspace/isort/tests/postgres_tests/test_array.py:after	2017-10-30 11:40:04.624306
@@ -11,6 +11,7 @@
 from django.test import TransactionTestCase, modify_settings, override_settings
 from django.test.utils import isolate_apps
 from django.utils import timezone
+
 from . import PostgreSQLTestCase, PostgreSQLWidgetTestCase
 from .models import (

```

That newline you removed should be there 😉

**Comment by mpasternak** [comment]
@adamchainz CI says thing's legit now, thank you for your suggestions.

**Pull request merged**